bot.py

```diff
@@ -251,119 +292,102 @@ def get_members_by_role(role):
         message_parts = message.content.split(' ', 1)
 
         if len(message_parts) == 1:
-            log.info("[{0}] No game specified".format(member.name))
-            msg = "{user}, you didn't specify a game, are you retarded?"
+            log.info("[{0}] No game/role specified".format(member.name))
+            msg = "{user}, you didn't specify a game/role, are you retarded?"
             await client.send_message(message.channel, msg.format(user=member.mention))
             return
 
-        game = message_parts[1].upper()
+        role = message_parts[1]
+        #Legacy support for old geezers
+        if role == "shitposting":
+            role = "shitty_people"
+        if role == "promote_a_stream":
+            role = "creeps"
 
-        if game in valid_games:
-            log.info("[{user}] Requested to leave game: {game}".format(user=member.name, game=game))
+        if role.upper() in valid_games or role.lower() in valid_roles:
+            log.info("[{user}] requested to leave game/role: {role}".format(user=member.name, role=role))
 
             # Check to see if the user has this role
             for author_role in member.roles:
-                if author_role.name == game:
+                if author_role.name == role:
                     # They did, so remove the role
                     await client.remove_roles(member, author_role)
                     log.info("[{0}] Role removed".format(member))
-                    msg = "Hello {user}, you have been removed from list for {game}, to re-join send `!{game}` in any channel."
-                    await client.send_message(message.channel, msg.format(user=member.mention, game=game))
-
-                    notification_msg = "{user} no longer wants to play {game} like a bitch. There are currently {players} players available."
-                    await client.send_message(get_channel("looking_for_group"), notification_msg.format(user=member.mention, game=game, players=len(get_members_by_role(game))))
+                    if role in valid_games:
+                        reply = "Hello {user}, you have been removed from list for {role}, to re-join send `!join {role}` in any channel.".format(user=member.mention, role=role)
+                        notification_msg = "{user} no longer wants to play {role} like a bitch. There are currently {players} players available."
+                        await client.send_message(get_channel("looking_for_group"), notification_msg.format(user=member.mention, role=role, players=len(get_members_by_role(role))))
+                    elif role == "shitty_people":
+                        reply = "Alright {user}, I see how it is. {channel} is now off-limits to you.".format(user=member.mention, channel=get_channel("nsfw_shitposting").mention)
+                    elif role == "creeps":
+                        reply = "Ok {user}, you will no longer be notified of live streams.".format(user=member.mention)
```

Comment:
There should be an `else` here as well with a generic "you have been added to this role" message.